src/library/scala/concurrent/Future.scala

```diff
@@ -513,80 +527,116 @@ object Future {
     classOf[Unit]    -> classOf[scala.runtime.BoxedUnit]
   )
 
+  private[this] final val _cachedId: AnyRef => AnyRef = Predef.identity _
+
+  private[concurrent] final def id[T]: T => T = _cachedId.asInstanceOf[T => T]
+
+  private[concurrent] final val collectFailed =
+    (t: Any) => throw new NoSuchElementException("Future.collect partial function is not defined at: " + t) with NoStackTrace
+
+  private[concurrent] final val filterFailure =
+    Failure[Nothing](new NoSuchElementException("Future.filter predicate is not satisfied") with NoStackTrace)
+
+  private[this] final val failedFailure =
+    Failure[Nothing](new NoSuchElementException("Future.failed not completed with a throwable.") with NoStackTrace)
+
+  private[concurrent] final val failedFailureFuture: Future[Nothing] =
+    scala.concurrent.Future.fromTry(failedFailure)
+
+  private[this] final val _failedFun: Try[Any] => Try[Throwable] =
+    v => if (v.isInstanceOf[Failure[Any]]) Success(v.asInstanceOf[Failure[Any]].exception) else failedFailure
+
+  private[concurrent] final def failedFun[T]: Try[T] => Try[Throwable] = _failedFun.asInstanceOf[Try[T] => Try[Throwable]]
+
+  private[concurrent] final val recoverWithFailedMarker: Future[Nothing] =
+    scala.concurrent.Future.failed(new Throwable with NoStackTrace)
```

Comment:
This should use the Throwable [constructor](https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#Throwable-java.lang.String-java.lang.Throwable-boolean-boolean-) which disables suppressed exceptions and writeable stacktraces.